Objective of the Research:

The objective of my research is to uncover insights about the evolution of the business aspect of the movie industry. Understanding audience preferences, and helping writers and directors with the best possible portrayal of their story and imagination on the screen is a vital focus of any company and producer. Being able to make data-driven decisions and picking up market trends can make significant changes in the revenue stream and profit.

The key focuses of my analysis are the following:

Genre

Writer

Director

Budget

Profit

In [22]:
# First let's import the packages we will use in this project
# You can do this all now or as you need them
import pandas as pd
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt
import matplotlib.mlab as mlab
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
pd.options.mode.chained_assignment = None




# Now we need to read in the data
# Now we need to read in the data
data = pd.read_csv('/content/movies.csv')


i. Glimpse of the Data

In [23]:
data.head(10)

name rating      genre  year  \
0                                     The Shining      R      Drama  1980   
1                                 The Blue Lagoon      R  Adventure  1980   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
3                                       Airplane!     PG     Comedy  1980   
4                                      Caddyshack      R     Comedy  1980   
5                                 Friday the 13th      R     Horror  1980   
6                              The Blues Brothers      R     Action  1980   
7                                     Raging Bull      R  Biography  1980   
8                                     Superman II     PG     Action  1980   
9                                 The Long Riders      R  Biography  1980   

                            released  score      votes            director  \
0      June 13, 1980 (United States)    8.4   927000.0     Stanley Kubrick   
1       July 2, 1980 (United States)    5.8    65000.0      Randal Kleiser   
2      June 20, 1980 (United States)    8.7  1200000.0      Irvin Kershner   
3       July 2, 1980 (United States)    7.7   221000.0        Jim Abrahams   
4      July 25, 1980 (United States)    7.3   108000.0        Harold Ramis   
5        May 9, 1980 (United States)    6.4   123000.0  Sean S. Cunningham   
6      June 20, 1980 (United States)    7.9   188000.0         John Landis   
7  December 19, 1980 (United States)    8.2   330000.0     Martin Scorsese   
8      June 19, 1981 (United States)    6.8   101000.0      Richard Lester   
9       May 16, 1980 (United States)    7.0    10000.0         Walter Hill   

                    writer             star         country      budget  \
0             Stephen King   Jack Nicholson  United Kingdom  19000000.0   
1  Henry De Vere Stacpoole   Brooke Shields   United States   4500000.0   
2           Leigh Brackett      Mark Hamill   United States  18000000.0   
3             Jim Abrahams      Robert Hays   United States   3500000.0   
4       Brian Doyle-Murray      Chevy Chase   United States   6000000.0   
5            Victor Miller     Betsy Palmer   United States    550000.0   
6              Dan Aykroyd     John Belushi   United States  27000000.0   
7             Jake LaMotta   Robert De Niro   United States  18000000.0   
8             Jerry Siegel     Gene Hackman   United States  54000000.0   
9              Bill Bryden  David Carradine   United States  10000000.0   

         gross                       company  runtime  
0   46998772.0                  Warner Bros.    146.0  
1   58853106.0             Columbia Pictures    104.0  
2  538375067.0                     Lucasfilm    124.0  
3   83453539.0            Paramount Pictures     88.0  
4   39846344.0                Orion Pictures     98.0  
5   39754601.0            Paramount Pictures     95.0  
6  115229890.0            Universal Pictures    133.0  
7   23402427.0  Chartoff-Winkler Productions    129.0  
8  108185706.0                Dovemead Films    127.0  
9   15795189.0                United Artists    100.0

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7668 entries, 0 to 7667
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   name      7668 non-null   object 
 1   rating    7591 non-null   object 
 2   genre     7668 non-null   object 
 3   year      7668 non-null   int64  
 4   released  7666 non-null   object 
 5   score     7665 non-null   float64
 6   votes     7665 non-null   float64
 7   director  7668 non-null   object 
 8   writer    7665 non-null   object 
 9   star      7667 non-null   object 
 10  country   7665 non-null   object 
 11  budget    5497 non-null   float64
 12  gross     7479 non-null   float64
 13  company   7651 non-null   object 
 14  runtime   7664 non-null   float64
dtypes: float64(5), int64(1), object(9)
memory usage: 898.7+ KB


ii. Knowing of the Data:

In [25]:
data.shape

(7668, 15)

Obeservation: The dataset comprises 7668 rows and 15 columns. The rows are information about different movies that were released from 1980 to 2020. The columns provide valuable information about each of the movies.

In [26]:
# Columns of the data
data.columns

Index(['name', 'rating', 'genre', 'year', 'released', 'score', 'votes',
       'director', 'writer', 'star', 'country', 'budget', 'gross', 'company',
       'runtime'],
      dtype='object')

Obeservation: The dataset comprises 7668 rows and 15 columns. The rows are information about different movies that were released from 1980 to 2020. The columns provide valuable information about each of the movies.

Obeservation: The columns of the dataset provide valuable information about each movie.

name: Name of the Movie

rating: Rating of the movie (R, PG, etc.)

genre: Main Genre of the movie

year: Year of Release

released: Release Date(YYYY-MM-DD)(location)

score: IMDb user rating

votes: Number of user votes

director: Name of the Director

writer: Name of the Writer

star: Main Actor/Actress

country: Country of Origin

budget: Budget of the movie

gross: Revenue of the movie

company: The Production Company

runtime: Duration of the movie in minutes

In [27]:
data["genre"].unique()

array(['Drama', 'Adventure', 'Action', 'Comedy', 'Horror', 'Biography',
       'Crime', 'Fantasy', 'Family', 'Sci-Fi', 'Animation', 'Romance',
       'Music', 'Western', 'Thriller', 'History', 'Mystery', 'Sport',
       'Musical'], dtype=object)

Obeservation: There are 19 unique genres of movies in the dataset.

Obeservation: Most of the missing values in the dataset are observed in the budget column, followed by gross and some values from other columns.

Decision: For the sake of our analysis, we shall remove all the missing values.

Now, Let's Get a Quick Overview

In [17]:
print(f"Records: {data.shape[0]}")
print(f"Features: {data.shape[1]}")

Records: 7668
Features: 15


In [28]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7668 entries, 0 to 7667
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   name      7668 non-null   object 
 1   rating    7591 non-null   object 
 2   genre     7668 non-null   object 
 3   year      7668 non-null   int64  
 4   released  7666 non-null   object 
 5   score     7665 non-null   float64
 6   votes     7665 non-null   float64
 7   director  7668 non-null   object 
 8   writer    7665 non-null   object 
 9   star      7667 non-null   object 
 10  country   7665 non-null   object 
 11  budget    5497 non-null   float64
 12  gross     7479 non-null   float64
 13  company   7651 non-null   object 
 14  runtime   7664 non-null   float64
dtypes: float64(5), int64(1), object(9)
memory usage: 898.7+ KB


In [29]:
# Calculating the missing values in different columns
data.isnull().sum().sort_values(ascending=False)

budget      2171
gross        189
rating        77
company       17
runtime        4
score          3
votes          3
writer         3
country        3
released       2
star           1
name           0
genre          0
year           0
director       0
dtype: int64

Obeservation: Most of the missing values in the dataset are observed in the budget column, followed by gross and some values from other columns.

Decision: For the sake of our analysis, we shall remove all the missing values.

In [30]:
# Removing Missing Values

dataFrame.isnull().sum().sort_values(ascending = False)

name        0
rating      0
genre       0
year        0
released    0
score       0
votes       0
director    0
writer      0
star        0
country     0
budget      0
gross       0
company     0
runtime     0
dtype: int64

In [21]:
dataFrame.shape

(5421, 15)

Obeservation: After removing the missing values, we are left with information about 5421 movies.

iii. Introducing New Features

In [31]:
dataFrame["profit"] = dataFrame["gross"] - dataFrame["budget"]
dataFrame['result'] = dataFrame['profit'].apply(lambda x: 'hit' if x > 0 else 'flop')
dataFrame.head(10)

name rating      genre  year  \
0                                     The Shining      R      Drama  1980   
1                                 The Blue Lagoon      R  Adventure  1980   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
3                                       Airplane!     PG     Comedy  1980   
4                                      Caddyshack      R     Comedy  1980   
5                                 Friday the 13th      R     Horror  1980   
6                              The Blues Brothers      R     Action  1980   
7                                     Raging Bull      R  Biography  1980   
8                                     Superman II     PG     Action  1980   
9                                 The Long Riders      R  Biography  1980   

                            released  score      votes            director  \
0      June 13, 1980 (United States)    8.4   927000.0     Stanley Kubrick   
1       July 2, 1980 (United States)    5.8    65000.0      Randal Kleiser   
2      June 20, 1980 (United States)    8.7  1200000.0      Irvin Kershner   
3       July 2, 1980 (United States)    7.7   221000.0        Jim Abrahams   
4      July 25, 1980 (United States)    7.3   108000.0        Harold Ramis   
5        May 9, 1980 (United States)    6.4   123000.0  Sean S. Cunningham   
6      June 20, 1980 (United States)    7.9   188000.0         John Landis   
7  December 19, 1980 (United States)    8.2   330000.0     Martin Scorsese   
8      June 19, 1981 (United States)    6.8   101000.0      Richard Lester   
9       May 16, 1980 (United States)    7.0    10000.0         Walter Hill   

                    writer             star         country      budget  \
0             Stephen King   Jack Nicholson  United Kingdom  19000000.0   
1  Henry De Vere Stacpoole   Brooke Shields   United States   4500000.0   
2           Leigh Brackett      Mark Hamill   United States  18000000.0   
3             Jim Abrahams      Robert Hays   United States   3500000.0   
4       Brian Doyle-Murray      Chevy Chase   United States   6000000.0   
5            Victor Miller     Betsy Palmer   United States    550000.0   
6              Dan Aykroyd     John Belushi   United States  27000000.0   
7             Jake LaMotta   Robert De Niro   United States  18000000.0   
8             Jerry Siegel     Gene Hackman   United States  54000000.0   
9              Bill Bryden  David Carradine   United States  10000000.0   

         gross                       company  runtime       profit result  
0   46998772.0                  Warner Bros.    146.0   27998772.0    hit  
1   58853106.0             Columbia Pictures    104.0   54353106.0    hit  
2  538375067.0                     Lucasfilm    124.0  520375067.0    hit  
3   83453539.0            Paramount Pictures     88.0   79953539.0    hit  
4   39846344.0                Orion Pictures     98.0   33846344.0    hit  
5   39754601.0            Paramount Pictures     95.0   39204601.0    hit  
6  115229890.0            Universal Pictures    133.0   88229890.0    hit  
7   23402427.0  Chartoff-Winkler Productions    129.0    5402427.0    hit  
8  108185706.0                Dovemead Films    127.0   54185706.0    hit  
9   15795189.0                United Artists    100.0    5795189.0    hit

Obeservation: Since we are focused on the business side of things, we calculated profit from the gross revenue and budget and also classified films as hit/flop based on the profit

Analyzing on Various Factors of the Data:




i. Based on Score:

In [32]:
#by score
highest_rated_movies = dataFrame.groupby('year').apply(lambda x: x.loc[x['score'].idxmax()])
highest_rated_movies.head(10)

name rating      genre  year  \
year                                                                           
1980  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
1981   Indiana Jones and the Raiders of the Lost Ark     PG     Action  1981   
1982                                    Blade Runner      R     Action  1982   
1983                                        Scarface      R      Crime  1983   
1984                     Once Upon a Time in America      R      Crime  1984   
1985                              Back to the Future     PG  Adventure  1985   
1986                                          Aliens      R     Action  1986   
1987                               Full Metal Jacket      R      Drama  1987   
1988                                 Cinema Paradiso     PG      Drama  1988   
1989              Indiana Jones and the Last Crusade  PG-13     Action  1989   

                               released  score      votes            director  \
year                                                                            
1980      June 20, 1980 (United States)    8.7  1200000.0      Irvin Kershner   
1981      June 12, 1981 (United States)    8.4   905000.0    Steven Spielberg   
1982      June 25, 1982 (United States)    8.1   710000.0        Ridley Scott   
1983   December 9, 1983 (United States)    8.3   766000.0      Brian De Palma   
1984       June 1, 1984 (United States)    8.4   321000.0        Sergio Leone   
1985       July 3, 1985 (United States)    8.5  1100000.0     Robert Zemeckis   
1986      July 18, 1986 (United States)    8.3   668000.0       James Cameron   
1987      July 10, 1987 (United States)    8.3   691000.0     Stanley Kubrick   
1988  February 23, 1990 (United States)    8.5   239000.0  Giuseppe Tornatore   
1989       May 24, 1989 (United States)    8.2   707000.0    Steven Spielberg   

                  writer              star         country      budget  \
year                                                                     
1980      Leigh Brackett       Mark Hamill   United States  18000000.0   
1981     Lawrence Kasdan     Harrison Ford   United States  18000000.0   
1982     Hampton Fancher     Harrison Ford   United States  28000000.0   
1983        Oliver Stone         Al Pacino   United States  25000000.0   
1984          Harry Grey    Robert De Niro           Italy  30000000.0   
1985     Robert Zemeckis    Michael J. Fox   United States  19000000.0   
1986       James Cameron  Sigourney Weaver  United Kingdom  18500000.0   
1987     Stanley Kubrick    Matthew Modine  United Kingdom  30000000.0   
1988  Giuseppe Tornatore   Philippe Noiret           Italy   5000000.0   
1989        Jeffrey Boam     Harrison Ford   United States  48000000.0   

            gross                company  runtime       profit result  
year                                                                   
1980  538375067.0              Lucasfilm    124.0  520375067.0    hit  
1981  389925971.0     Paramount Pictures    115.0  371925971.0    hit  
1982   41676878.0       The Ladd Company    117.0   13676878.0    hit  
1983   65884703.0     Universal Pictures    170.0   40884703.0    hit  
1984    5473212.0       The Ladd Company    229.0  -24526788.0   flop  
1985  381906762.0     Universal Pictures    116.0  362906762.0    hit  
1986  131060248.0  Twentieth Century Fox    137.0  112560248.0    hit  
1987   46357676.0                 Natant    116.0   16357676.0    hit  
1988   13019063.0          Cristaldifilm    155.0    8019063.0    hit  
1989  474171806.0     Paramount Pictures    127.0  426171806.0    hit

In [33]:
hit_count = len(highest_rated_movies[highest_rated_movies['result']=="hit"])
flop_count = len(highest_rated_movies[highest_rated_movies['result'] == "flop"])


print(f"Number of Hit Films: {hit_count}")
print(f"Number of Flop Films: {flop_count}")

Number of Hit Films: 40
Number of Flop Films: 1


Obeservation: This subset has the highest rated movies over the years. We can also observe although one of the movies was highest rated of a particular year, it still was flop based on the revenue generation.

ii. Based on Budget:

In [34]:
# by budget
movies_with_highest_budget = dataFrame.groupby('year').apply(lambda x: x.loc[x['budget'].idxmax()])
movies_with_highest_budget.head(10)

name rating      genre  year  \
year                                                    
1980              Superman II     PG     Action  1980   
1981                   Inchon     PG      Drama  1981   
1982                    Annie     PG     Comedy  1982   
1983                    Krull     PG     Action  1983   
1984          The Cotton Club      R      Crime  1984   
1985   Santa Claus: The Movie     PG  Adventure  1985   
1986                  Pirates  PG-13     Action  1986   
1987                   Ishtar  PG-13     Action  1987   
1988  Who Framed Roger Rabbit     PG  Animation  1988   
1989                The Abyss  PG-13  Adventure  1989   

                                released  score     votes  \
year                                                        
1980       June 19, 1981 (United States)    6.8  101000.0   
1981  September 17, 1982 (United States)    2.7     659.0   
1982       June 18, 1982 (United States)    6.6   41000.0   
1983       July 29, 1983 (United States)    6.1   31000.0   
1984   December 14, 1984 (United States)    6.5   17000.0   
1985   November 27, 1985 (United States)    6.3   12000.0   
1986       July 18, 1986 (United States)    6.1    8100.0   
1987        May 15, 1987 (United States)    4.5   11000.0   
1988       June 22, 1988 (United States)    7.7  187000.0   
1989      August 9, 1989 (United States)    7.5  167000.0   

                  director            writer              star  \
year                                                             
1980        Richard Lester      Jerry Siegel      Gene Hackman   
1981         Terence Young       Robin Moore  Laurence Olivier   
1982           John Huston    Carol Sobieski      Aileen Quinn   
1983           Peter Yates  Stanford Sherman      Ken Marshall   
1984  Francis Ford Coppola   William Kennedy      Richard Gere   
1985        Jeannot Szwarc      David Newman      Dudley Moore   
1986        Roman Polanski    John Brownjohn    Walter Matthau   
1987            Elaine May        Elaine May     Warren Beatty   
1988       Robert Zemeckis      Gary K. Wolf       Bob Hoskins   
1989         James Cameron     James Cameron         Ed Harris   

             country      budget        gross                  company  \
year                                                                     
1980   United States  54000000.0  108185706.0           Dovemead Films   
1981     South Korea  46000000.0    5200986.0      One Way Productions   
1982   United States  50000000.0   57063861.0        Columbia Pictures   
1983  United Kingdom  47000000.0   16916617.0        Columbia Pictures   
1984   United States  58000000.0   25928721.0         Zoetrope Studios   
1985  United Kingdom  50000000.0   23717291.0                      GGG   
1986          France  40000000.0    1641825.0  Carthago Films S.a.r.l.   
1987   United States  55000000.0   14375181.0        Columbia Pictures   
1988   United States  70000000.0  329803958.0      Touchstone Pictures   
1989   United States  70000000.0   90000098.0    Twentieth Century Fox   

      runtime       profit result  
year                               
1980    127.0   54185706.0    hit  
1981    140.0  -40799014.0   flop  
1982    127.0    7063861.0    hit  
1983    121.0  -30083383.0   flop  
1984    127.0  -32071279.0   flop  
1985    107.0  -26282709.0   flop  
1986    121.0  -38358175.0   flop  
1987    107.0  -40624819.0   flop  
1988    104.0  259803958.0    hit  
1989    140.0   20000098.0    hit

In [35]:
hit_count = len(movies_with_highest_budget[movies_with_highest_budget['result']=="hit"])
flop_count = len(movies_with_highest_budget[movies_with_highest_budget['result'] == "flop"])


print(f"Number of Hit Films: {hit_count}")
print(f"Number of Flop Films: {flop_count}")

Number of Hit Films: 35
Number of Flop Films: 6


Obeservation: This subset has the movies with the highest budget over the years. We can also observe that 6 of the 35 films, turned out to be flop although they had a massive budget compared to the other films that came out on that particular year.

iii. Based on Votes:

In [36]:
# by votes
movies_with_highest_votes = dataFrame.groupby('year').apply(lambda x: x.loc[x['votes'].idxmax()])
movies_with_highest_votes.head(10)

name rating      genre  year  \
year                                                                           
1980  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
1981   Indiana Jones and the Raiders of the Lost Ark     PG     Action  1981   
1982                                    Blade Runner      R     Action  1982   
1983      Star Wars: Episode VI - Return of the Jedi     PG     Action  1983   
1984                                  The Terminator      R     Action  1984   
1985                              Back to the Future     PG  Adventure  1985   
1986                                          Aliens      R     Action  1986   
1987                               Full Metal Jacket      R      Drama  1987   
1988                                        Die Hard      R     Action  1988   
1989              Indiana Jones and the Last Crusade  PG-13     Action  1989   

                              released  score      votes          director  \
year                                                                         
1980     June 20, 1980 (United States)    8.7  1200000.0    Irvin Kershner   
1981     June 12, 1981 (United States)    8.4   905000.0  Steven Spielberg   
1982     June 25, 1982 (United States)    8.1   710000.0      Ridley Scott   
1983      May 25, 1983 (United States)    8.3   973000.0  Richard Marquand   
1984  October 26, 1984 (United States)    8.0   812000.0     James Cameron   
1985      July 3, 1985 (United States)    8.5  1100000.0   Robert Zemeckis   
1986     July 18, 1986 (United States)    8.3   668000.0     James Cameron   
1987     July 10, 1987 (United States)    8.3   691000.0   Stanley Kubrick   
1988     July 20, 1988 (United States)    8.2   810000.0    John McTiernan   
1989      May 24, 1989 (United States)    8.2   707000.0  Steven Spielberg   

               writer                   star         country      budget  \
year                                                                       
1980   Leigh Brackett            Mark Hamill   United States  18000000.0   
1981  Lawrence Kasdan          Harrison Ford   United States  18000000.0   
1982  Hampton Fancher          Harrison Ford   United States  28000000.0   
1983  Lawrence Kasdan            Mark Hamill   United States  32500000.0   
1984    James Cameron  Arnold Schwarzenegger  United Kingdom   6400000.0   
1985  Robert Zemeckis         Michael J. Fox   United States  19000000.0   
1986    James Cameron       Sigourney Weaver  United Kingdom  18500000.0   
1987  Stanley Kubrick         Matthew Modine  United Kingdom  30000000.0   
1988   Roderick Thorp           Bruce Willis   United States  28000000.0   
1989     Jeffrey Boam          Harrison Ford   United States  48000000.0   

            gross                company  runtime       profit result  
year                                                                   
1980  538375067.0              Lucasfilm    124.0  520375067.0    hit  
1981  389925971.0     Paramount Pictures    115.0  371925971.0    hit  
1982   41676878.0       The Ladd Company    117.0   13676878.0    hit  
1983  475106177.0              Lucasfilm    131.0  442606177.0    hit  
1984   78371200.0             Cinema '84    107.0   71971200.0    hit  
1985  381906762.0     Universal Pictures    116.0  362906762.0    hit  
1986  131060248.0  Twentieth Century Fox    137.0  112560248.0    hit  
1987   46357676.0                 Natant    116.0   16357676.0    hit  
1988  141603197.0  Twentieth Century Fox    132.0  113603197.0    hit  
1989  474171806.0     Paramount Pictures    127.0  426171806.0    hit

In [37]:
hit_count = len(movies_with_highest_votes[movies_with_highest_votes['result']=="hit"])
flop_count = len(movies_with_highest_votes[movies_with_highest_votes['result'] == "flop"])


print(f"Number of Hit Films: {hit_count}")
print(f"Number of Flop Films: {flop_count}")

Number of Hit Films: 41
Number of Flop Films: 0


Obeservation: These movies can be considered as the highest watched movies over the years. We observe that all of them successful on the terms of profit generation.

iv. Based on Gross Revenue:

In [38]:
# by gross revenue
movies_with_highest_gross_revenue = dataFrame.groupby('year').apply(lambda x: x.loc[x['gross'].idxmax()])
movies_with_highest_gross_revenue.head(10)

name rating      genre  year  \
year                                                                           
1980  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
1981   Indiana Jones and the Raiders of the Lost Ark     PG     Action  1981   
1982                      E.T. the Extra-Terrestrial     PG     Family  1982   
1983      Star Wars: Episode VI - Return of the Jedi     PG     Action  1983   
1984            Indiana Jones and the Temple of Doom     PG     Action  1984   
1985                              Back to the Future     PG  Adventure  1985   
1986                                         Top Gun     PG     Action  1986   
1987                                Fatal Attraction      R      Drama  1987   
1988                                        Rain Man      R      Drama  1988   
1989              Indiana Jones and the Last Crusade  PG-13     Action  1989   

                                released  score      votes          director  \
year                                                                           
1980       June 20, 1980 (United States)    8.7  1200000.0    Irvin Kershner   
1981       June 12, 1981 (United States)    8.4   905000.0  Steven Spielberg   
1982       June 11, 1982 (United States)    7.8   381000.0  Steven Spielberg   
1983        May 25, 1983 (United States)    8.3   973000.0  Richard Marquand   
1984        May 23, 1984 (United States)    7.5   459000.0  Steven Spielberg   
1985        July 3, 1985 (United States)    8.5  1100000.0   Robert Zemeckis   
1986        May 16, 1986 (United States)    6.9   306000.0        Tony Scott   
1987  September 18, 1987 (United States)    6.9    79000.0       Adrian Lyne   
1988   December 16, 1988 (United States)    8.0   483000.0    Barry Levinson   
1989        May 24, 1989 (United States)    8.2   707000.0  Steven Spielberg   

                writer             star        country      budget  \
year                                                                 
1980    Leigh Brackett      Mark Hamill  United States  18000000.0   
1981   Lawrence Kasdan    Harrison Ford  United States  18000000.0   
1982  Melissa Mathison     Henry Thomas  United States  10500000.0   
1983   Lawrence Kasdan      Mark Hamill  United States  32500000.0   
1984     Willard Huyck    Harrison Ford  United States  28000000.0   
1985   Robert Zemeckis   Michael J. Fox  United States  19000000.0   
1986          Jim Cash       Tom Cruise  United States  15000000.0   
1987     James Dearden  Michael Douglas  United States  14000000.0   
1988      Barry Morrow   Dustin Hoffman  United States  25000000.0   
1989      Jeffrey Boam    Harrison Ford  United States  48000000.0   

            gross             company  runtime       profit result  
year                                                                
1980  538375067.0           Lucasfilm    124.0  520375067.0    hit  
1981  389925971.0  Paramount Pictures    115.0  371925971.0    hit  
1982  792910554.0  Universal Pictures    115.0  782410554.0    hit  
1983  475106177.0           Lucasfilm    131.0  442606177.0    hit  
1984  333107271.0  Paramount Pictures    118.0  305107271.0    hit  
1985  381906762.0  Universal Pictures    116.0  362906762.0    hit  
1986  357288178.0  Paramount Pictures    110.0  342288178.0    hit  
1987  320145693.0  Paramount Pictures    119.0  306145693.0    hit  
1988  354825435.0      United Artists    133.0  329825435.0    hit  
1989  474171806.0  Paramount Pictures    127.0  426171806.0    hit

In [39]:
hit_count = len(movies_with_highest_gross_revenue[movies_with_highest_gross_revenue['result']=="hit"])
flop_count = len(movies_with_highest_gross_revenue[movies_with_highest_gross_revenue['result'] == "flop"])


print(f"Number of Hit Films: {hit_count}")
print(f"Number of Flop Films: {flop_count}")

Number of Hit Films: 41
Number of Flop Films: 0


Obeservation: This subset has contains movies with the highest gross revenue over the years. All of them were economically successful as it was expected

v. Based on Profit:

In [40]:
# by profit
highest_profit_generating_movies = dataFrame.groupby('year').apply(lambda x: x.loc[x['profit'].idxmax()])
highest_profit_generating_movies.head(10)

name rating      genre  year  \
year                                                                           
1980  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
1981   Indiana Jones and the Raiders of the Lost Ark     PG     Action  1981   
1982                      E.T. the Extra-Terrestrial     PG     Family  1982   
1983      Star Wars: Episode VI - Return of the Jedi     PG     Action  1983   
1984            Indiana Jones and the Temple of Doom     PG     Action  1984   
1985                              Back to the Future     PG  Adventure  1985   
1986                                         Top Gun     PG     Action  1986   
1987                                Fatal Attraction      R      Drama  1987   
1988                                        Rain Man      R      Drama  1988   
1989              Indiana Jones and the Last Crusade  PG-13     Action  1989   

                                released  score      votes          director  \
year                                                                           
1980       June 20, 1980 (United States)    8.7  1200000.0    Irvin Kershner   
1981       June 12, 1981 (United States)    8.4   905000.0  Steven Spielberg   
1982       June 11, 1982 (United States)    7.8   381000.0  Steven Spielberg   
1983        May 25, 1983 (United States)    8.3   973000.0  Richard Marquand   
1984        May 23, 1984 (United States)    7.5   459000.0  Steven Spielberg   
1985        July 3, 1985 (United States)    8.5  1100000.0   Robert Zemeckis   
1986        May 16, 1986 (United States)    6.9   306000.0        Tony Scott   
1987  September 18, 1987 (United States)    6.9    79000.0       Adrian Lyne   
1988   December 16, 1988 (United States)    8.0   483000.0    Barry Levinson   
1989        May 24, 1989 (United States)    8.2   707000.0  Steven Spielberg   

                writer             star        country      budget  \
year                                                                 
1980    Leigh Brackett      Mark Hamill  United States  18000000.0   
1981   Lawrence Kasdan    Harrison Ford  United States  18000000.0   
1982  Melissa Mathison     Henry Thomas  United States  10500000.0   
1983   Lawrence Kasdan      Mark Hamill  United States  32500000.0   
1984     Willard Huyck    Harrison Ford  United States  28000000.0   
1985   Robert Zemeckis   Michael J. Fox  United States  19000000.0   
1986          Jim Cash       Tom Cruise  United States  15000000.0   
1987     James Dearden  Michael Douglas  United States  14000000.0   
1988      Barry Morrow   Dustin Hoffman  United States  25000000.0   
1989      Jeffrey Boam    Harrison Ford  United States  48000000.0   

            gross             company  runtime       profit result  
year                                                                
1980  538375067.0           Lucasfilm    124.0  520375067.0    hit  
1981  389925971.0  Paramount Pictures    115.0  371925971.0    hit  
1982  792910554.0  Universal Pictures    115.0  782410554.0    hit  
1983  475106177.0           Lucasfilm    131.0  442606177.0    hit  
1984  333107271.0  Paramount Pictures    118.0  305107271.0    hit  
1985  381906762.0  Universal Pictures    116.0  362906762.0    hit  
1986  357288178.0  Paramount Pictures    110.0  342288178.0    hit  
1987  320145693.0  Paramount Pictures    119.0  306145693.0    hit  
1988  354825435.0      United Artists    133.0  329825435.0    hit  
1989  474171806.0  Paramount Pictures    127.0  426171806.0    hit

Obeservation: This subset contains movies with the highest profit over the years.

Visualizing different Aspects and Trends:

i. Genre Distribution:

In [41]:
def create_pie_chart_data(subset, title):
    genre_counts = subset['genre'].value_counts()
    fig = go.Figure(data=[go.Pie(labels=genre_counts.index, values=genre_counts.values, textinfo='percent+label')])
    fig.update_layout(title_text=title)
    return fig



fig1 = create_pie_chart_data(highest_rated_movies, "Genre Distribution of Highest Score")
fig2 = create_pie_chart_data(movies_with_highest_budget, "Genre Distribution of Highest Budget")
fig3 = create_pie_chart_data(movies_with_highest_votes, "Genre Distribution of Highest Vote ")
fig4 = create_pie_chart_data(highest_profit_generating_movies, "Genre Distribution of Highest Profit ")


fig = make_subplots(rows=2, cols=2, subplot_titles=["Highest Score", "Highest Budget", "Highest Vote", "Highest Profit"],specs=[[{"type": "pie"}, {"type": "pie"}],
           [{"type": "pie"}, {"type": "pie"}]],
)
fig.add_trace(fig1.data[0], row=1, col=1)
fig.add_trace(fig2.data[0], row=1, col=2)
fig.add_trace(fig3.data[0], row=2, col=1)
fig.add_trace(fig4.data[0], row=2, col=2)

fig.update_layout(width=1000, height=800,title="Genre Distribution", showlegend=False)
fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")
fig.show()

Obeservation:The pie chart helps us to understand industry preferences and audience preferences for different genres of movies.

73.2% of the movies with the most budget are Action movies, followed by Adventure with 9.76% and Animation with 7.32%.

Action movies are also the ones with the most numbers of economically successful films(63.4%). Animation(14.6%) comes in second followed by Drama(9.76%).


A similar gap in numbers is not observed in the case of Scoring. Action and Drama come really close with 36.6% and 22% followed by Crime at 17.1%, Biography at 9.76%, and Adventure at 7.32%.

But the presence of Action in Voting is again noteworthy.51.2% followed by Crime, Drama, Adventure, and Biography at 17.1%, 14.6%, 9.76%, and 7.32% respectively.

ii. Top Directors Over the Years:

In [42]:
def create_bar_plot_data(subset, column_name):
    top_counts = subset[column_name].value_counts().nlargest(10)
    colors = px.colors.qualitative.Set3
    color_mapping = dict(zip(top_counts.index, colors))
    fig = go.Figure(data=[go.Bar(x=top_counts.index, y=top_counts.values)])
    fig.update_layout(xaxis_title=column_name, yaxis_title='Count')
    top_counts_color = top_counts.index.map(color_mapping)
    fig = px.bar(top_counts, x=top_counts.index, y=top_counts.values)
    fig.update_traces(marker=dict(color=top_counts_color))
    return fig


fig1 = create_bar_plot_data(highest_rated_movies, 'director')
fig2 = create_bar_plot_data(movies_with_highest_budget, 'director')
fig3 = create_bar_plot_data(movies_with_highest_votes, 'director')
fig4 = create_bar_plot_data(highest_profit_generating_movies, 'director')


fig = make_subplots(rows=4, cols=1, subplot_titles=["Highest Scored Films", "Highest Budgeted Films", "Highest Voted Films", "Highest Profit Films"])
fig.add_trace(fig1.data[0], row=1, col=1)
fig.add_trace(fig2.data[0], row=2, col=1)
fig.add_trace(fig3.data[0], row=3, col=1)
fig.add_trace(fig4.data[0], row=4, col=1)


fig.update_layout(width=1000,
                height=1200,
                title="Top Directors over the Years",
                showlegend=False)


fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")

fig.show()

Obeservation: There are some interesting facts that can be observed when analyzing the performance of the directors of the years. For our analysis we focused on the peak performance only, i.e how many times they have come out at top.

Christopher Nolan: Nolan takes the first position with the Highest Scored Films subseta and the Highest Voted Films subset. His films were had the highest rating for five times and highest voted for six times. Budget wise he came fifth with two instances. Although his films were economically successful but they didn't break any records.

James Cameron: James Cameron is a director who always prefers high budet films. Budget wise he came at the top with four instances. Both in terms of profit and voting, he came third with 3 appearences.

Steven Spielberg: In our analysis the most interesting fact that came to surface is that Steven Spielberg himself is a strong candidate as the Director. In terms of profitability he came out at top with five instance. In terms of voting he came second as well as scoring. Both were 4 instances

iii. Top Writers Over the Years:

In [43]:
fig1 = create_bar_plot_data(highest_rated_movies, 'writer')
fig2 = create_bar_plot_data(movies_with_highest_budget, 'writer')
fig3 = create_bar_plot_data(movies_with_highest_votes, 'writer')
fig4 = create_bar_plot_data(highest_profit_generating_movies, 'writer')

# Create subplots with four small bar plots
fig = make_subplots(rows=4, cols=1, subplot_titles=["Highest Scored Films", "Highest Budgeted Films", "Highest Voted Films", "Highest Profit Films"])
fig.add_trace(fig1.data[0], row=1, col=1)
fig.add_trace(fig2.data[0], row=2, col=1)
fig.add_trace(fig3.data[0], row=3, col=1)
fig.add_trace(fig4.data[0], row=4, col=1)

# Set the main plot width and height to 800x600
fig.update_layout(width=1000, height=1200, title="Top writers Across Subsets", showlegend=False)
fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")
fig.show()

Obeservation: James Cameron came out as the undisputed choice as the writer.

3 times in Budget, Votes and Profit. A perfect combination.

iv. Highest Film Budgets over the Years:

In [44]:
fig = px.line(movies_with_highest_budget, x='year', y='budget', title='Highest Film Budgets over the Years')
fig.update_layout(width=1000, height=800,xaxis_title='Year', yaxis_title='Budget')
fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")
fig.update_traces(mode='lines+markers', marker_line_width=2, marker_size=10)
fig.show()

Obeservation: The budget flow for movies have changed drastically after 1990.

Before 1990 the range was from 54-70m USD
After 1990 it rose 175m which reached its peak at 2019 with 356m USD

v. Profit Generatinon over the Years:

In [45]:
fig = px.line(movies_with_highest_budget, x='year', y='profit', title='Highest Profit over the Years')
fig.update_layout(width=1000, height=800,xaxis_title='Year', yaxis_title='Profit Generatinon over the Years')
fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")
fig.update_traces(mode='lines+markers', marker_line_width=2, marker_size=10)
fig.show()

Obeservation:

Two huge spikes of 2B USD and 2.44B USD is
observed in the year 1997 and 2019.

The overall profit margin has increased over the years from 54m to 1.5B USD

Covid had affected the movie industry thus having a low profit of 158m in 2020.

vi. Run Time Breakdown For Profit Generating Genres:

In [46]:
fig = px.scatter(highest_profit_generating_movies, x="profit", y="runtime", color="genre", symbol="genre",
                 size='votes', hover_data=['runtime'], title="Run Time Breakdown For Profit Generating Genres")
fig.update_layout(width=1000, height=800,xaxis_title='Profit($/USD)', yaxis_title='Run Time (minutes)')
fig.update_layout(
    margin=dict(l=40, r=40, t=40, b=40),
    paper_bgcolor="LightSteelBlue")
fig.show()

Obeservation: Run time varies a lot for profit generating movies of different genres.

Animation movies have a range 80-103 minutes

Family movies have a range of 115-129

Drama movies have a range 119-133. But an exception is also observed with a runtime of 194 minutes

Adventure movies have a range of 115-152 minutes

Action movies have the most largest range and variation. Starting from 110 to 201 minutes.

Key Insights:
Action movies dominate the genre space in all fields: Highest Budget, Highest Vote, Highest Score, Highest Profit

Nolan, Cameron and Spielberg himself, are the directors to look out for.

Cameron in terms of film writing, is one the of the undisputed guy in the field.

Lack of Female personalities in cases of Direction/Writing was observed.

Fluctuation of budget is observed after 1990.

Two massive spikes in the figures of 2 and 2.44 Billion USD is observed in 1997 and 2019.

Covid had affected the movie industry thus having a low profit of 158m in 2020.